gmail/modules/listener/Dispatcher.bal

```diff
@@ -16,175 +16,215 @@
         changedLabel.messageDetail = message;
-        check callOnEmailLabelAdded(self.httpService, changedLabel);
+        check self.adaptor.callOnEmailLabelAdded(changedLabel);
     }
 
     isolated function dispatchStarredEmail(gmail:HistoryEvent addedlabel) returns @tainted error? {
         if (addedlabel?.labelIds is string[]) {
             foreach var label in <string[]>addedlabel?.labelIds {
-                match label{
-                    STARRED =>{
-                        gmail:Message message = check self.gmailClient->readMessage(<@untainted>addedlabel.message.id);
-                        check callOnEmailStarred(self.httpService, message);
+                match label {
+                    STARRED => {
+                        gmail:Message message = check self.readMessage(<@untainted>addedlabel.message.id);
+                        check self.adaptor.callOnEmailStarred(message);
                     }
                 }
             }
-        }        
+        }
     }
 
-    isolated function dispatchLabelRemovedEmail(gmail:HistoryEvent removedLabel) returns @tainted error?{
-        ChangedLabel changedLabel = { messageDetail: {id : "", threadId : ""}, changedLabelId: []};
+    isolated function dispatchLabelRemovedEmail(gmail:HistoryEvent removedLabel) returns @tainted error? {
+        ChangedLabel changedLabel = {messageDetail: {id: "", threadId: ""}, changedLabelId: []};
         if (removedLabel?.labelIds is string[]) {
             changedLabel.changedLabelId = <string[]>removedLabel?.labelIds;
         }
-        gmail:Message message = check self.gmailClient->readMessage(<@untainted>removedLabel.message.id);
+        gmail:Message message = check self.readMessage(<@untainted>removedLabel.message.id);
         changedLabel.messageDetail = message;
-        check callOnEmailLabelRemoved(self.httpService, changedLabel);
+        check self.adaptor.callOnEmailLabelRemoved(changedLabel);
     }
 
     isolated function dispatchStarRemovedEmail(gmail:HistoryEvent removedLabel) returns @tainted error? {
         if (removedLabel?.labelIds is string[]) {
             foreach var label in <string[]>removedLabel?.labelIds {
-                match label{
-                    STARRED =>{
-                        gmail:Message message = check self.gmailClient->readMessage(<@untainted>removedLabel.message.id);
-                        check callOnEmailStarRemoved(self.httpService, message);
+                match label {
+                    STARRED => {
+                        gmail:Message message = check self.readMessage(<@untainted>removedLabel.message.id);
+                        check self.adaptor.callOnEmailStarRemoved(message);
                     }
                 }
             }
-        }        
+        }
+    }
+
+    isolated function readMessage(string messageId, string? format = (), string[]? metadataHeaders = (), 
```

Comment:
Does this method belong here? Is it related to dispatching messages? Seems it is doing something else.

Reply:
It was a utility method used by dispatch related functions.
Now i have moved them out.